documentation/jetty-documentation/src/main/asciidoc/programming-guide/server/websocket/server-websocket-filter.adoc

```diff
@@ -0,0 +1,79 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+[[pg-server-websocket-configure-filter]]
+==== Advanced `WebSocketUpgradeFilter` Configuration
+
+The WebSocket ``ServletContainerInitializer``s, namely `JavaxWebSocketServletContainerInitializer` and `JettyWebSocketServletContainerInitializer`, install the `WebSocketUpgradeFilter` to handle HTTP requests that upgrade to WebSocket.
```

Comment:
This is not correct. 

The jetty SCI doesn't add the filter, rather it is added in the addMapping method on the container. Only the javax SCI always adds the filter.

Moreover,  the SCIs do a lot more than just install the filter:  pools for inflaters, deflaters, buffers; executors; object factory; mappings; contains; lifecycle listeners etc.   

 The SCIs initialize a lot of machinery and much of it is extensible/configurable to the users so probably should be of note in the "Advanced" documentation.    So I think we should open with a full enumeration of the mechanism that need to be initialized and how/where they are.